images/controller/internal/controllers/rvr_tie_breaker_count/reconciler_test.go

```diff
@@ -0,0 +1,535 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rvrtiebreakercount_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/client/interceptor"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	v1alpha1 "github.com/deckhouse/sds-replicated-volume/api/v1alpha1"
+	"github.com/deckhouse/sds-replicated-volume/api/v1alpha3"
+	rvrtiebreakercount "github.com/deckhouse/sds-replicated-volume/images/controller/internal/controllers/rvr_tie_breaker_count"
+)
+
+var expectedError = errors.New("test error")
+
+var _ = Describe("Reconcile", func() {
+	scheme := runtime.NewScheme()
+	Expect(corev1.AddToScheme(scheme)).To(Succeed())
+	Expect(v1alpha1.AddToScheme(scheme)).To(Succeed())
+	Expect(v1alpha3.AddToScheme(scheme)).To(Succeed())
+
+	var (
+		builder *fake.ClientBuilder
+		cl      client.WithWatch
+		rec     *rvrtiebreakercount.Reconciler
+	)
+
+	BeforeEach(func() {
+		builder = fake.NewClientBuilder().WithScheme(scheme)
+		cl = nil
+		rec = nil
+	})
+
+	JustBeforeEach(func() {
+		cl = builder.Build()
+		rec = rvrtiebreakercount.NewReconciler(cl, logr.New(log.NullLogSink{}), scheme)
+	})
+
+	It("returns nil when ReplicatedVolume not found", func(ctx SpecContext) {
+		result, err := rec.Reconcile(ctx, reconcile.Request{NamespacedName: client.ObjectKey{Name: "non-existent"}})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(result).To(Equal(reconcile.Result{}))
+	})
+
+	When("rv created", func() {
+		var rv v1alpha3.ReplicatedVolume
+		BeforeEach(func() {
+			rv = v1alpha3.ReplicatedVolume{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "rv1",
+				},
+				Spec: v1alpha3.ReplicatedVolumeSpec{
+					ReplicatedStorageClassName: "rsc1",
+				},
+			}
+		})
+
+		JustBeforeEach(func(ctx SpecContext) {
+			Expect(cl.Create(ctx, &rv)).To(Succeed())
+		})
+
+		When("ReplicatedStorageClassName is empty", func() {
+			BeforeEach(func() {
+				rv.Spec.ReplicatedStorageClassName = ""
+			})
+
+			It("returns nil when ReplicatedStorageClassName is empty", func(ctx SpecContext) {
+				result, err := rec.Reconcile(ctx, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&rv)})
+				Expect(err).NotTo(HaveOccurred())
+				Expect(result).To(Equal(reconcile.Result{}))
```

Comment:
```suggestion
				Expect(rec.Reconcile(ctx, reconcile.Request{NamespacedName: client.ObjectKeyFromObject(&rv)})).Error().NotTo(HaveOccurred())
```

Пофиг на reconcile.Result{}